### ISSUE TITLE
bug: metamask mobile browser cannot connect after disconnecting

### ISSUE THREAD

**Issue by tea-juuni** [wagmi-dev/wagmi#1365]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package Version

0.7.2

### Current Behavior

You can reproduce this bug in metamask mobile browser on the demo site wagmi.sh. Basically once you disconnect and try to reconnect to metamask you will face this error

```
The method wallet_requestPermissions does not exist/is not available
```

This only happens on metamask mobile and goes away after you clear privacy data.

### Expected Behavior

Disconnect/Connect should work without issues on metamask mobile browser

### Steps To Reproduce

- use metmask mobile browser (ios or android)
- visit wagmi.sh
- click connect
- click disconnect
- click connect again, see bug

### Link to Minimal Reproducible Example (CodeSandbox, StackBlitz, etc.)

N/A

### Anything else?

N/A

**Comment by tea-juuni** [comment]
Might be fixed by latest, will try 0.7.5 and report back

**Comment by tea-juuni** [comment]
Tested and still running into issue on 0.7.5, haven't tried 0.8.6 due to my project not supporting ESM-only packages but may try a fresh app later to test
